components/DocumentListTabs/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { Dispatch, ReactNode, SetStateAction } from 'react'
+import Tabs, { TabsProps } from '@mui/material/Tabs'
+import Tab from '@mui/material/Tab'
+import Box from '@mui/system/Box'
+import Person from '@mui/icons-material/Person'
+import Drafts from '@mui/icons-material/Drafts'
+import UploadFile from '@mui/icons-material/UploadFile'
+import Lock from '@mui/icons-material/Lock'
+
+// This enum should be moved into models, as soon as that is ready.
+// There is a bug in eslint, it makes enum members as unused. The workaround is more annoying.
+// Instead we can just disable and re-enable the rule.
+// ref: https://stackoverflow.com/questions/57802057/eslint-configuring-no-unused-vars-for-typescript
+
+/* eslint-disable no-unused-vars */
+export enum DocumentStatus {
+  All,
+  Draft,
+  Published,
+  Archived,
+}
+/* eslint-enable no-unused-vars */
```

Comment:
i went to look at the link you inserted, and tried to add the typescript plugin and did not face this error. maybe we could add that? because having this disable and enable is quite ugly